imap_processing/tests/glows/test_glows_l1a_cdf.py

```diff
@@ -32,9 +33,21 @@ def l1a_data():
     return (histograms_l1a, de_l1a)
 
 
+def test_create_glows_attr_obj():
+    glows_attrs = create_glows_attr_obj()
+    glows_his_attrs = glows_attrs.get_variable_attributes("histogram_attrs")
+    for variable_attrs in glows_attrs.variable_attribute_schema["attribute_key"]:
+        required_var_attributes = glows_attrs.variable_attribute_schema[
+            "attribute_key"
+        ][variable_attrs]["required"]
+        if required_var_attributes is True:
+            assert variable_attrs in glows_his_attrs.keys()
+
+
 def test_generate_histogram_dataset(l1a_data):
     histograms_l1a, _ = l1a_data
-    dataset = generate_histogram_dataset(histograms_l1a, "v001")
+    glows_attrs = create_glows_attr_obj()
```

Comment:
Ah, I see now why you made that a method. In this case, I'll leave it up to you whether you want to leave it as a method or just duplicate in the tests. I know I said "Don't Repeat Yourself" but it's usually ok to do in unit tests because it makes it easier to change small pieces (like, if I wanted to add a new variable attribute file or something in the unit tests.)

Sorry, I feel like so much of my advice is contradictory.... IMO coding is an art, not a science, so some of it is style preferences, and some of it is to move towards my ultimate goal to have clean, beautiful, logical code. Like poetry... you want to take away all the words that are unnecessary and only leave the things that are necessary.

Reply:
This is wonderful advice! Yes, coding is definitely an art. I was thinking about this, and while normally I would vote on the side of taking it out as a method, I actually think it works well in this case. We use it in both `generate_de_dataset`, and `generate_histogram_dataset`, we are able to load in global and variable attributes in one place, and we are able to load in the `data_version` directly to this method. Thoughts?

Reply:
Up to you, your logic makes sense to me